src/Dfe.Spi.GraphQlApi.Infrastructure.SearchApi/SearchApiSearchProvider.cs

```diff
@@ -19,7 +20,12 @@ public SearchApiSearchProvider(IRestClient restClient, SearchConfiguration confi
         {
             _restClient = restClient;
             _restClient.BaseUrl = new Uri(configuration.SearchApiBaseUrl, UriKind.Absolute);
-            
+            if (!string.IsNullOrEmpty(configuration.SearchApiFunctionKey))
+            {
+                _restClient.DefaultParameters.Add(new Parameter("x-functions-key", configuration.SearchApiFunctionKey,
```

Comment:
Will work for the demo, but same comment re: generic headers. Whether it saves you time now or later, I don't mind really.

Reply:
We can look at something generic later. We will need it in a bunch of the apps.